### PR TITLE
Replaces deprecated SqlBlock with optimized version

### PR DESCRIPTION
**Pull request by RCHowell**
## Relevant Issues

#1560 

## Description

This replaces the deprecated SqlDialect and SqlBlock with the fixed versions.

## Other Information
- Updated Unreleased Section in CHANGELOG: **NO**
- Any backward-incompatible changes? **NO**
- Any new external dependencies? **NO**
- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? **YES**

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-ast/src/main/kotlin/org/partiql/ast/sql/SqlLayout.kt [hunk 1]
```diff
@@ -1,97 +1,78 @@
...
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+
 package org.partiql.ast.sql
 
 /**
  * [SqlLayout] determines how an [SqlBlock] tree is transformed in SQL text.
  */
-@Deprecated("This will be changed in the next major version")
```
Comment (alancai98) [inline partiql-ast/src/main/kotlin/org/partiql/ast/sql/SqlLayout.kt:6]: Given that we're changing the `SqlBlock`, `SqlDialect`, and `SqlLayout` public APIs, I think we should include an entry in the CHANGELOG for the 0.15 release.

**Review by alancai98 — COMMENTED** [review]
Replacement of the previous AST printing looks good. Just one suggestion to update the CHANGELOG since we're working off of `main`.

File: partiql-ast/src/main/kotlin/org/partiql/ast/sql/SqlLayout.kt (hunk 1 above)
Comment (RCHowell) [reply partiql-ast/src/main/kotlin/org/partiql/ast/sql/SqlLayout.kt:6]: ack

**Review by alancai98 — APPROVED** [review]
nice, glad we can remove some more deprecated code

**Pull request merged**